administrator/components/com_media/resources/scripts/components/upload/upload.vue

```diff
@@ -0,0 +1,67 @@
+<template>
+    <input type="file" class="hidden"
+           :name="name"
+           :multiple="multiple"
+           :accept="accept"
+           @change="upload"
+           ref="fileInput">
+</template>
+<script>
+    export default {
+        name: 'media-upload',
+        props: {
+            accept: {
+                type: String,
+            },
+            extensions: {
+                default: () => [],
+            },
+            name: {
+                type: String,
+                default: 'file',
+            },
+            multiple: {
+                type: Boolean,
+                default: true,
+            },
+        },
+        methods: {
+            /* Open the choose-file dialog */
+            chooseFiles() {
+                this.$refs['fileInput'].click();
+            },
+            /* Upload files */
+            upload(e) {
+                e.preventDefault();
+                const files = e.target.files;
+
+                // Loop through array of files and upload each file
+                for (let file of files) {
+
+                    // Create a new file reader instance
+                    let reader = new FileReader();
+
+                    // Add the on load callback
+                    reader.onload = (progressEvent) => {
+                        const result = progressEvent.target.result,
+                            splitIndex = result.indexOf('base64') + 7,
```

Comment:
I was not 100% sure about this part... It is something that could be easily done on server-side also.